src/compiler/compile/Component.ts

```diff
@@ -1,3 +1,6 @@
+// TODO: remove after https://github.com/jridgewell/trace-mapping/pull/24 has been merged
+// @ts-expect-error Svelte 3's ancient TS version can't look up the types for this package properly yet
```

Comment:
If you update to the latest version of `@jridgewell/trace-mapping` it can now resolve the types. I went ahead and merged the latest `master` into this PR so that you can see any type errors as previously it would crash rather than show the error messages
```suggestion
```

Reply:
svelte-4 branch has been started, but once that's usable for this the merge conflicts would be so large it's going to be easier to just start a new PR instead of this one.  cc @dummdidumm